lib/plan.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Braintree.Plan do
+  @moduledoc """
+  Get information about braintree plans
+
+  For additional reference see:
+  https://developers.braintreepayments.com/reference/request/plan/all/ruby
+  """
+
+  alias Braintree.HTTP
+  alias Braintree.ErrorResponse, as: Error
+
+  @doc """
+  Get all existing plans or a error response.
+
+  ## Example
+
+      {:ok, plans} = Braintree.Plan.all()
+
+      plans # All the existing plans
+  """
+  def all do
+    case HTTP.get("plans") do
+      {:ok, %{"plans" => plans}} ->
+        plans
```

Comment:
This will return a list of maps, but won't turn them into structs. For consistency, we need a `Plan` struct, with type definitions, and a constructor. You can see examples of the type definition, type specs, and constructors in some of the other modules.